cg/store/api/find_business_data.py

```diff
@@ -462,30 +462,54 @@ def get_ready_made_library_expected_reads(self, case_id: str) -> int:
             )
         return application.expected_reads
 
-    def get_all_samples(self) -> List[Sample]:
+    def get_samples(self) -> List[Sample]:
         """Return all samples."""
         return self._get_query(table=Sample).order_by(Sample.created_at.desc()).all()
 
-    def get_samples_by_enquiry(
-        self, *, customers: Optional[List[Customer]] = None, enquiry: str = None
+    def get_samples_by_customer_id(
+        self, *, customers: Optional[List[Customer]] = None
     ) -> List[Sample]:
-        records = self._get_query(table=Sample)
+        """Return all the pools for a customer."""
+        customer_ids = [customer.id for customer in customers]
+        return apply_sample_filter(
+            samples=self._get_query(table=Sample),
+            customer_ids=customer_ids,
+            filter_functions=[SampleFilter.FILTER_BY_CUSTOMER_ID],
+        ).all()
 
-        if customers:
-            customer_ids = [customer.id for customer in customers]
-            records = records.filter(Sample.customer_id.in_(customer_ids))
+    def get_samples_by_name_enquiry(self, name_enquiry: str) -> List[Sample]:
+        """Return all samples that match name enquiry."""
+        return apply_sample_filter(
+            samples=self._get_query(table=Sample),
+            name_enquiry=name_enquiry,
+            filter_functions=[SampleFilter.FILTER_BY_NAME_ENQUIRY],
+        ).all()
 
-        records = (
-            records.filter(
-                or_(
-                    Sample.name.like(f"%{enquiry}%"),
-                    Sample.internal_id.like(f"%{enquiry}%"),
+    def get_samples_by_order_enquiry(self, order_enquiry: str) -> List[Sample]:
+        """Return all that match the order enquiry."""
+        return apply_sample_filter(
+            samples=self._get_query(table=Sample),
+            order_enquiry=order_enquiry,
+            filter_functions=[SampleFilter.FILTER_BY_ORDER_ENQUIRY],
+        ).all()
```

Comment:
Many of these functions are quite obscure to me, could you trace them to heir usage and see if they are still relevant? It would be nice to be able to remove some of them

Reply:
yes we can do that in another PR.